modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/LiferayApp.js

```diff
@@ -18,11 +18,43 @@ import LiferaySurface from '../surface/Surface';
 import {getPortletBoundaryId, getUid, resetAllPortlets} from '../util/utils';
 import App from './App';
 
-import '../css/lfr_spa_loading_bar.scss';
-
 const MAX_TIMEOUT = Math.pow(2, 31) - 1;
 const PROPAGATED_PARAMS = ['bodyCssClass'];
 
+const CSS = `
```

Comment:
:thinking: Maybe we can create a bundler plugin for this. Similar to webpack's plugin that returns the CSS as a text, instead of placing it directly in the HTML...

It would be something like this:

```javascript
import css from '../css/lfr_spa_loading_bar.scss';
      .
      .
      .		
document.head.appendChild(
    buildFragment(`<style type="text/css">${CSS}</style>`)
);
```

It could even return the fragment directly, if we wanted to...

Reply:
Yeah that might be a good idea because the file we end up requesting is doing that exact same thing.

I don't expect to do this manually in many other places right now though. I really only did it in this instance because it was a pretty small set of CSS and our spa module is fairly unique to other apps in DXP. I wouldn't expect other apps to add CSS manually like this.

Reply:
I'll test this idea locally and see